### PR TITLE
add profile::core::letsencrypt

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

File: hieradata/site/dev.yaml [hunk 1]
```diff
@@ -103,3 +103,5 @@ accounts::user_list:
 # Stop iptables by default - the default rules are highly restrictive to the
 # point of harm and we don't have a meaningful permission set to make this useful.
 firewall::ensure: "stopped"
+
+letsencrypt::server: "https://acme-staging.api.letsencrypt.org/directory"  # testing url
```
Comment (hreinking) [inline hieradata/site/dev.yaml:107]: I know is a better practice, but you plan to merge the "staging" server? Also,  I thought the server was https://acme-v02.api.letsencrypt.org/directory

File: site/profile/manifests/core/letsencrypt.pp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      '/root/.aws/credentials':
+        ensure  => file,
+        mode    => '0600',
+        backup  => false,
+        content => $aws_credentials,
+      ;
+    }
+
+    # aws credentials required by dns_route53 plugin.
+    File['/root/.aws/credentials'] -> Letsencrypt::Certonly<| |>
```
Comment (hreinking) [inline site/profile/manifests/core/letsencrypt.pp:60]: From where are you fetching the aws credentials?

Comment (hreinking) [reply site/profile/manifests/core/letsencrypt.pp:60]: never mind, just checked the private

File: hieradata/site/dev.yaml (hunk 1 above)
Comment (jhoblitt) [reply hieradata/site/dev.yaml:107]: Yes, this only applies to the dev site.

**Pull request merged**